### PR TITLE
Structured Concurrency in Java

### PR DESCRIPTION
**Pull request by rcardin**
Hey, man. I finally did it. Here is the first version of the article on structured concurrency in Java.
Hope you like it!

Let me know if anything needs to be clarified.

### CODE REVIEW

File: src/content/articles/structured-concurrency-in-java.mdx [hunk 1]
```diff
@@ -0,0 +1,1609 @@
...
+
+## Conclusion
+
+We finally reach the conclusions of the article. We introduced what structured concurrency is and what its benefits are. We saw it's hard to avoid thread leaks using the traditional Java concurrency API. Fortunately, Project Loom is available in version 19 of Java. The project introduced virtual threads and structured concurrency. We saw the different shutdown policies that the JDK brings, and we implemented one custom policy to dive deep into the main concepts of structured concurrency. We were able to implement structured concurrency primitives that we find in many libraries like Scala ZIO and Cats Effects. Finally, we saw the parent-children relationship in structured concurrency in action and how the `close` method is the last line of defense to ensure that all the children tasks that are not needed anymore are interrupted.
```
Comment (daniel-ciocirlan) [inline src/content/articles/structured-concurrency-in-java.mdx:1524]: "Cats Effect"

**Review by daniel-ciocirlan — COMMENTED** [review]
Amazing piece 👏 
After a few hiccups at the beginning replicating what you've done, the flow was super smooth, and the explanations are top notch 👌 

A few comments (mostly nits), then we can merge.
I have ideas.

File: src/content/articles/structured-concurrency-in-java.mdx [hunk 2]
```diff
@@ -0,0 +1,1609 @@
+---
+author: riccardo-cardin
+category: guide
+title: "Project Loom: Structured Concurrency in Java"
+excerpt: "Concurrency is a complex topic that requires a deep understanding of the underlying mechanisms of the programming language and the runtime environment. Fortunately, Project Loom is here to help. In this article, we will explore the concept of structured concurrency and how it simplifies writing concurrent code in Java."
```
[the remark this answers is not in the document]
Comment (andreilgeorgescu) [reply src/content/articles/structured-concurrency-in-java.mdx:5]: It has to be a single sentence that doesn't end in a period with the goal being to succinctly get the goal across to people browsing the website. You can write a compound sentence with a colon. Emdash isn't preferred.

**Comment by daniel-ciocirlan** [comment]
@rcardin please test the article and the site locally. For every tag that doesn't exist in the current set, we need to add a new JSON file under `src/content/tags`. For this article it means

- `src/content/tags/loom.json`
- `src/content/tags/structured-concurrency.json`

You can keep the JSONs empty. We have JSONs because we might need to add metadata later.

Also, for this article I recommend we use the tag `concurrency` instead of `structured-concurrency`.

File: src/content/articles/structured-concurrency-in-java.mdx [hunk 3]
```diff
@@ -0,0 +1,1609 @@
...
+    LOGGER.info("Repositories found for user '{}'", userId);
+    return List.of(
+        new Repository(
+                "raise4s", Visibility.PUBLIC, URI.create("https://github.com/rcardin/raise4s")),
+        new Repository(
+                "sus4s", Visibility.PUBLIC, URI.create("https://github.com/rcardin/sus4s")));
+  }
+}
+
+private static void delay(Duration duration) throws InterruptedException {
```
[the remark this answers is not in the document]
Comment (rcardin) [reply src/content/articles/structured-concurrency-in-java.mdx:106]: I changed the scope.

File: src/content/articles/structured-concurrency-in-java.mdx [hunk 4]
```diff
@@ -0,0 +1,1609 @@
...
+  final GitHubUser gitHubUser = service.findGitHubUser(new UserId(1L));
+
+  LOGGER.info("GitHub user: {}", gitHubUser);
+}
+```
+
+The output of the execution of the above code is the following:
+
+```
+08:50:34.159 [virtual-22] INFO GitHubApp -- Finding repositories for user with id 'UserId[value=1]'
```
[the remark this answers is not in the document]
Comment (rcardin) [reply src/content/articles/structured-concurrency-in-java.mdx:215]: I don't have any special configuration of logback :screaming:

**Comment by rcardin** [comment]
@daniel-ciocirlan, I should have fixed all the issues. Please let me know if you see anything else.

**Review by daniel-ciocirlan — APPROVED** [review]

**Comment by daniel-ciocirlan** [comment]
@rcardin I'll merge this next week when I have the video ready so that we can publish both at the same time.
Well done 👏

**Comment by daniel-ciocirlan** [comment]
Checks haven't triggered and I don't know why, but I tried this locally, and so I'll bypass the checks

**Pull request merged**